_posts/2025-04-19-navigating-pagination-with-hybrid-query.md

```diff
@@ -0,0 +1,229 @@
+
+Modifying `from` or ``size`` values increases the number of results retrieved for each subquery from a shard. However, these new results don't simply append to the last page. They can appear earlier in the ranked list, changing the ground truth you're trying to paginate through. This happens in the following two main scenarios:
+
+1. **Higher-weighted subquery effect**:
+    - New results from a higher-weighted subquery may rank higher after score combination.
+    - As a result, these new entries may appear earlier in the final result set---either on the first few pages or somewhere in the middle---rather than at the end, disrupting the expected pagination order.
+
+    In our example, increasing size to 4 causes an unexpected result: the new Query 1 result (document ID 14) appears as the fourth document instead of at the end, because of Query 1's higher weight during normalization, as shown in the following diagram.
+
+    ![more-weightage](/assets/media/blog-images/2025-04-19-navigating-pagination-with-hybrid-query/more-weightage.png){:class="img-centered"}
+
+1. **Duplicate result effect**:
+    - Newly retrieved results can become duplicates, meaning they exist in both subquery results.
+    - After the score combination process, the combined score of a duplicate entry increases, making it more relevant.
+    - With equal subquery weights, duplicate results rank higher than expected, appearing earlier in search results.
+
+    In the following diagram, documents with IDs 0 and 12 are duplicate results appearing in both match and k-NN subqueries. Because duplicate documents typically receive higher hybrid scores than single-occurrence documents, they appear higher in the results, disrupting the expected pagination order.
+
+    ![duplicate-results](/assets/media/blog-images/2025-04-19-navigating-pagination-with-hybrid-query/duplicate-results.png){:class="img-centered"}
+
+### Addressing the ground truth problem
+
+The `pagination_depth` parameter solves these issues by ensuring a consistent number of results per shard. This removes the dependency on from and size during result retrieval, ensuring that the same set of search results every time. The `from` and `size` parameters are applied after hybrid techniques to trim the results and determine how many results should be displayed on each page.
+
+### Key takeaways
+
+Note the following key takeaways:
+
+* Keep `pagination_depth` constant while navigating between pages to maintain consistent results.
+* Changing `pagination_depth` during pagination will lead to inconsistent results.
+
+In our earlier example, setting `pagination_depth = 5` addresses the ground truth issue. Both match query and k-NN query will return up to 5 results from each shard to the coordinator node so it can apply hybrid techniques to the subquery results. The `from` and `size` parameters then determine how to trim the hybrid search results, as shown in the following diagram.
```

Comment:
```suggestion
In our earlier example, setting `pagination_depth = 5` addresses the ground truth issue. Both the match and k-NN queries will return up to 5 results from each shard to the coordinator node so that it can apply hybrid techniques to the subquery results. The `from` and `size` parameters then determine how to trim the hybrid search results, as shown in the following diagram.
```